vnext/ReactUWP/Views/DynamicAutomationPeer.cpp

```diff
@@ -0,0 +1,190 @@
+
+  switch (accessibilityRole)
+  {
+  case winrt::react::uwp::AccessibilityRoles::Button:
+  case winrt::react::uwp::AccessibilityRoles::ImageButton:
+    return winrt::AutomationControlType::Button;
+  case winrt::react::uwp::AccessibilityRoles::Link:
+    return winrt::AutomationControlType::Hyperlink;
+  case winrt::react::uwp::AccessibilityRoles::Image:
+    return winrt::AutomationControlType::Image;
+  case winrt::react::uwp::AccessibilityRoles::KeyboardKey:
+    return winrt::AutomationControlType::Custom;
+  case winrt::react::uwp::AccessibilityRoles::Text:
+  case winrt::react::uwp::AccessibilityRoles::Summary:
+  case winrt::react::uwp::AccessibilityRoles::Header:
+    return winrt::AutomationControlType::Text;
+  case winrt::react::uwp::AccessibilityRoles::Adjustable:
+    return winrt::AutomationControlType::Slider;
+  case winrt::react::uwp::AccessibilityRoles::Search:
+  case winrt::react::uwp::AccessibilityRoles::Unknown:
+  default:
+    return winrt::AutomationControlType::Group;
+  }
+
+  return Super::GetAutomationControlTypeCore();
+}
+
+winrt::IInspectable DynamicAutomationPeer::GetPatternCore(winrt::PatternInterface const& patternInterface) const
+{
+  auto accessibilityRole = GetAccessibilityRole();
+
+  if (patternInterface == winrt::PatternInterface::Invoke &&
+    (accessibilityRole == winrt::react::uwp::AccessibilityRoles::Button || accessibilityRole == winrt::react::uwp::AccessibilityRoles::ImageButton))
+  {
+    return *this;
+  }
+  else if (patternInterface == winrt::PatternInterface::Selection || patternInterface == winrt::PatternInterface::SelectionItem)
+  {
+    return *this;
+  }
+
+  return Super::GetPatternCore(patternInterface);
+}
+
+bool DynamicAutomationPeer::IsEnabledCore() const
+{
+  bool disabled = GetAccessibilityState(winrt::react::uwp::AccessibilityStates::Disabled);
+  return !disabled && Super::IsEnabledCore();
+}
+
+// IInvokeProvider
+
+void DynamicAutomationPeer::Invoke() const
+{
+  if (auto invokeHandler = GetAccessibilityInvokeEventHandler())
+  {
+    invokeHandler();
+  }
+}
+
+// ISelectionProvider
+
+winrt::com_array<winrt::IRawElementProviderSimple> DynamicAutomationPeer::GetSelection() const
+{
+  // We don't differentiate between Views that are containers and Views that are items, and we don't
+  // have a way to specify the linkage between the two
+
+  // Returning nothing until that linkage exists
+  return {};
+}
+
+// ISelectionItemProvider
+
+bool DynamicAutomationPeer::IsSelected() const
+{
+  return GetAccessibilityState(winrt::react::uwp::AccessibilityStates::Selected);
+}
+
+winrt::IRawElementProviderSimple DynamicAutomationPeer::SelectionContainer() const
+{
+  // We don't differentiate between Views that are containers and Views that are items, and we don't
+  // have a way to specify the linkage between the two
+
+  // Returning nothing until that linkage exists
+
+  return nullptr;
+}
+
+void DynamicAutomationPeer::AddToSelection() const
+{
+  
```

Comment:
Are these not yet implemented?

Reply:
Right now RN does not have "selection" events, so this is a no-op